rs/sns/root/canister/canister.rs

```diff
@@ -29,14 +33,23 @@ use ic_sns_root::{
 };
 use icrc_ledger_types::icrc3::archive::ArchiveInfo;
 use prost::Message;
-use std::{cell::RefCell, time::Duration};
+use std::{
+    cell::RefCell,
+    time::{Duration, SystemTime},
+};
 
 type CanisterRuntime = CdkRuntime;
 const STABLE_MEM_BUFFER_SIZE: u32 = 100 * 1024 * 1024; // 100MiB
-const POLL_FOR_NEW_ARCHIVES_INTERVAL: Duration = Duration::from_secs(60 * 60 * 24); // one day
+
+const RUN_PERIODIC_TASKS_INTERVAL: Duration = Duration::from_secs(60 * 60 * 24); // one day
+
+/// This guarantees that timers cannot be restarted more often than once every 7 intervals.
+const RESET_TIMERS_COOL_DOWN_INTERVAL: Duration = Duration::from_secs(60 * 60 * 24 * 7); // one week
```

Comment:
This seems pretty long.

Reply:
I'm going to merge this as-is so that we can propose this today. I think having a reset once a week is better than not having it at all. We can adjust the constants later one.